vertexai/genai/tokenizer/corpora_test.go

```diff
@@ -0,0 +1,305 @@
+		{Pattern: regexp.MustCompile(`Abkhaz\-Cyrillic\+Abkh`), Encoding: charmap.Windows1251},
+	}
+
+	skip := map[string]bool{
+		"Burmese_Myanmar-UTF8":           true,
+		"Japanese_Nihongo-JIS":           true,
+		"Chinese_Mandarin-HZ":            true,
+		"Chinese_Mandarin-UTF8":          true,
+		"Gujarati-UTF8":                  true,
+		"Hungarian_Magyar-Unicode":       true,
+		"Lao-UTF8":                       true,
+		"Magahi-UTF8":                    true,
+		"Marathi-UTF8":                   true,
+		"Tamil-UTF8":                     true,
+		"Vietnamese-VPS":                 true,
+		"Vietnamese-VIQR":                true,
+		"Vietnamese-TCVN":                true,
+		"Magahi-Agra":                    true,
+		"Bhojpuri-Agra":                  true,
+		"Esperanto-T61":                  true,
+		"Burmese_Myanmar-WinResearcher":  true,
+		"Armenian-DallakHelv":            true,
+		"Tigrinya_Tigrigna-VG2Main":      true,
+		"Amharic-Afenegus6..60375":       true,
+		"Navaho_Dine-Navajo-Navaho-font": true,
+		"Azeri_Azerbaijani_Cyrillic-Az.Times.Cyr.Normal0117": true,
+		"Azeri_Azerbaijani_Latin-Az.Times.Lat0117":           true,
+		"Czech-Latin2-err":     true,
+		"Russian_Russky-UTF8~": true,
+	}
+
+	return &udhrCorpus{
+		Encodings: encodings,
+		Skip:      skip,
+	}
+}
+
+// getEncoding returns the encoding for a given filename based on patterns
+func (ucr *udhrCorpus) getEncoding(filename string) (encoding.Encoding, bool) {
+	for _, pattern := range ucr.Encodings {
+		if pattern.Pattern.MatchString(filename) {
+			return pattern.Encoding, true
+		}
+	}
+	return nil, false
+}
+
+// shouldSkip checks if the file should be skipped
+func (ucr *udhrCorpus) shouldSkip(filename string) bool {
+	return ucr.Skip[filename]
+}
+
+// decodeBytes converts a byte slice from a specified encoding to a UTF-8 string
+func decodeBytes(enc encoding.Encoding, data []byte) (string, error) {
+	if enc == encoding.Nop {
+		return string(data), nil
+	}
+
+	if enc == nil {
+		return "", fmt.Errorf("unsupported encoding or custom handling required")
+	}
+
+	decoder := enc.NewDecoder()
+	reader := transform.NewReader(strings.NewReader(string(data)), decoder)
+	decodedBytes, err := io.ReadAll(reader)
+	if err != nil {
+		return "", fmt.Errorf("error decoding data: %v", err)
+	}
+
+	return string(decodedBytes), nil
+}
+
+const defaultModel = "gemini-1.0-pro"
+const defaultLocation = "us-central1"
+
+func TestCountTokensWithCorpora(t *testing.T) {
+	projectID := os.Getenv("VERTEX_PROJECT_ID")
+	if testing.Short() {
+		t.Skip("skipping live test in -short mode")
+	}
+
+	if projectID == "" {
+		t.Skip("set a VERTEX_PROJECT_ID env var to run live tests")
+	}
+	ctx := context.Background()
+	client, err := genai.NewClient(ctx, projectID, defaultLocation)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer client.Close()
+	model := client.GenerativeModel(defaultModel)
+
+	t.Run("RemoteAndLocalCountTokensTest", func(t *testing.T) {
```

Comment:
Since this `TestCount...` function only runs this one test, you don't need `t.Run(...` (this is needed when you have sub-tests).

You can just place the test logic directly in the `TestCount...` function itself
Like here, for example: https://github.com/googleapis/google-cloud-go/blob/main/vertexai/genai/client_test.go#L357

Reply:
Done